### DISCUSSION TITLE
请问如何读取Resource中的SQLite数据库

### DISCUSSION THREAD

**Discussion by zeratulag** [BYJRK/DotNet-Discussions#59] [WPF 开发]
您好，我的 WPF .NET FrameWork项目中需要从一个小型SQLite数据库文件中读取数据（只读，大小仅有2M），假设这个数据库文件是a.db，我使用了sqlite-net包，则可以通过
`var db = new SQLiteConnection("a.db");`的方式建立连接。

现在我需要把a.db作为资源（将a.db包括在项目中，生成操作为Resource）放在我的WPF程序中一起发布（后续会使用Fody.Costura 将发布的项目打包成一个exe文件），此时a的路径变成了`pack://application:,,,/a.db`。
但是我无法用 `var db = new SQLiteConnection("pack://application:,,,/a.db");`的方式建立连接了，提示无效
SQLite.SQLiteException:“Could not open database file: pack://application:,,,/a.db (CannotOpen)”

在google上也没找到答案，请问应该如何正确读取？非常感谢！

**Comment by edward-chung** [comment]
Sqlite虽然提供了Mode=Memory创建专用的内存数据库的方式，但是并没有从Stream或者Byte[]中连接的方式。
[Sqlite内存中数据库](https://learn.microsoft.com/zh-cn/dotnet/standard/data/sqlite/in-memory-databases)
Resource生成操作会将文件编译成byte[]，所以应该不能这样连。
如果数据库中的内容需要加密的话可以考虑创建加密数据库，但是这个好像需要用到`System.Data.SQLite.SEE.License`，这玩意最低价$2000 USD。
但是可以考虑对数据库中的字段进行RSA加密

**Comment by dameng324** [comment]
如果数据库里的内容根本不会变化，为何还需要用sqlite数据库？直接把数据序列化成一个文件感觉更好一些，也更容易实现数据加密。想要实现从资源中加载Sqlite确实会比较麻烦。
